script/constructron.lua

```diff
@@ -799,46 +734,79 @@ me.conditions = {
     ---@param position MapPosition
     ---@return boolean
     position_done = function(job, position) -- this is condition for action "go_to_position"
-        local constructrons = job.constructrons
-        debug_lib.VisualDebugText("Moving to position", constructrons[1], -3, 1)
-        if constructrons[1].valid then
-            if (chunk_util.distance_between(constructrons[1].position, position) > 5) then -- the condition
-                if not ((job.attempt or 0) >= 3) then
-                    return false  -- condition is not met
-                else
-                    return "graceful_wrapup" -- unable to reach destination
-                end
+        local constructron = job.constructrons[1]
+        debug_lib.VisualDebugText("Moving to position", constructron, -3, 1)
+
+        local ticks = (game.tick - job.start_tick)
+        if ticks > 119 then return false end -- not enough time (two seconds) since last check
+
+        local distance_from_pos = chunk_util.distance_between(constructron.position, position)
+        if (distance_from_pos < 5) then return true end -- condition is satisfied
+
+        if not job.path_active then -- check if the path is active
+            if job.request_pathid and not global.pathfinder_requests[job.request_pathid] then -- check that there is a request
+                me.actions[job.action](job, table.unpack(job.action_args or {})) -- there is no request, request a path.
+            end
+            debug_lib.VisualDebugText("Path not active", constructron, 0.4, 3)
+            return false -- condition is not met
+        end
+
+        if job.landfill_job then -- is this a landfill job?
+            if not constructron.logistic_cell.logistic_network.can_satisfy_request("landfill", 1) then
+                -- !! Logic gap - Constructrons will return home even if there is other entities to build.

```

Comment:
Maybe instead of doing the graceful wrapup you could just skip to the next job in the bundle which would either be returning home or moving to another job site that could be a non landfill job.